### ISSUE TITLE
SA1000 incorrectly reported for default expressions

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#2420]
Currently SA1000 is getting reported for the following C# 7.1 syntax:

```csharp
var x = value == null ? default : 3;
```

**Comment by tillig** [comment]
Under .NET Core 3.1 and StyleCopAnalyzers 1.1.118 this appears to have been resolved (probably due to the submitted PR) and can be closed.

**Comment by sharwell** [comment]
It's still open to review cases where diagnostics _should_ be reported. Right now diagnostics are not reported even for unexpected spacing, such as when the spaces around `default` are removed.

**Comment by bjornhellander** [comment]
I think the current behaviour is correct. In this context, `default` is equivalent to a literal or a variable, so I expect SA1000 to continue to ignore it. Removing the trailing space in this example already triggers SA1003 and SA1024.

SA1000 tests should be added, though, to make sure that it doesn't regress.
